### ISSUE TITLE
Cannot mount nfs using cloud-config

### ISSUE THREAD

**Issue by kingsd041** [rancher/os#2606]
**RancherOS Version: (ros os version)**
1.5.0-rc2
**Where are you running RancherOS? (docker-machine, AWS, GCE, baremetal, etc.)**
virtualbox

Here is the issue from @bott0r

> Coming back to this. Without a nfs client i can just mount via a simple line:
`sudo mount -t nfs4 192.168.0.2:/mnt/volume01/data /mnt/data`
>
> However when i translate thet line to a cloudconfig with either:
>
> ```
> #/var/lib/rancher/conf/cloud-config.d/z.yml
> mounts:
> - ["192.168.0.2:/mnt/volume01/data", "/mnt/data", "nfs4", ""]
> ```
> or with
> 
> `ros config set mounts '[["192.168.0.2:/mnt/volume01/data", "/mnt/data1", "nfs4", ""]]'`
>
> it just doesnt work, what am i missing, i crawled by now through the documentation that is available or cached 404 pages, along side closed github issues. does anyone have a pointer here?

I reproduced this issue using rancheros v1.5.0-rc2 on virtualbox, and it works fine if I use rancheros v1.4.2.
I also tried to test mounts using v1.5.0-rc2 on vmware fusion and aws, all working fine.

**Comment by bott0r** [comment]
Hi, confused with your last sentence. Does it work on rc2 or you see the same error on rc2.

**Comment by kingsd041** [comment]
@bott0r  Both rc1 and rc2 have the same error. What environment do you run rancheros?

**Comment by bott0r** [comment]
bare metal and VM, both show the issue. Some people also report that it works fine for them.

**Comment by kingsd041** [comment]
I only reproduced this error on the virtual box.

**Comment by niusmallnan** [comment]
I tried 1.5.0-rc2 on virtualbox, and I can confirm that:
- mounting via a simple line can work
- using `ros mounts` cannot work

I checked the console logs when using `ros mounts`:
```
$ system-docker logs console
mount: mounting 172.22.100.101:/ on /mnt/nfs failed: Network is unreachable
time="2018-12-24T09:08:14Z" level=error msg="Failed to mount /mnt/nfs: exit status 255"
```
So why `Network is unreachable`? And v1.4.2 can work in this case?
ROS can configure the network in two places, the `cloud-init` service and the `network` service.
Usually, cloud-init starts very early. When `ros mounts` is executed, the system has obtained the network configuration. 
However, 1.5.0 introduces a new feature, when the user does not configure the cloud-init datasources, the cloud-init will not start, and the network will not be set in advance.
When ros mounts is running, there is a chance that the network service has not completed the network settings. Because by default we do not wait for the network service to complete, sometimes the network settings are slow, which will affect our system booting.

So if you can use this way, I think there will be no problem:
```
write_files:
- content: |+
    #!/bin/bash
    until ping -c1 172.22.100.101  &>/dev/null; do sleep 1; done
    mount -t nfs4 172.22.100.101:/ /mnt/nfs
  owner: root
  path: /etc/rc.local
  permissions: "0755"
```

**Comment by donnib** [comment]
How can this still be open and not fixed yet ? Is there a work around ?

**Comment by garethmorris78** [comment]
Its a real shame this has not been fixed still. I was looking forward to using RancherOS; really like the approach taken with the OS architecture. However the workaround listed is not appropriate for production environments.  In fact there seems to be a lack of documentation regarding mounting network storage in general which is a shame. 

Unless i'm wrong mounting network file share outside the cloud-init config would require a system container with nfs client doing the mounting ?

**Comment by ma-karai** [comment]
its really weird to me, as i feel its a super important feature 🗡

**Comment by cyrus104** [comment]
I would agree, I've spent more time trying to find this thread find a work around than it took me to get portainer and docker on a vm inside of proxmox.

I'm going to have to archive the VM for now. While I see a huge value in using this tool, I don't have the time to cobble something together and feel comfortable recommending this to go into production.

I will absolutely be monitoring this issue and as soon as a proper fix is in place so I can use the documented commands I fire it up for testing.

**Comment by zkhcohen** [comment]
+1

This has made RancherOS largely useless.

**Comment by ITJamie** [comment]
same. need nfs mounts in rancheros

**Comment by priceaj** [comment]
FYI this works for me on rancher/os:v1.5.8 running on HyperV on Windows10, NFS share served from QNAP NAS

Verify mount works correctly (`sudo mount -t nfs4 192.168.178.109:/Multimedia /NAS`)

`sudo ros service enable volume-nfs`

`sudo ros config set mounts '[["/dev/sdb1", "/ContainerConfig", "ext4", ""],["192.168.178.109:/Multimedia", "/NAS", "nfs4", ""]]'`

^ remember that set will overwrite NOT update existing config

`sudo reboot`

I realise that the original issue was on virtualbox... but there doesn't seem to be any documentation on how to do this anywhere!
